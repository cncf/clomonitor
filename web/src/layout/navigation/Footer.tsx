import { FaGithub } from 'react-icons/fa';
import { FiExternalLink } from 'react-icons/fi';

import logo from '../../media/clomonitor.svg';
import ExternalLink from '../common/ExternalLink';
import styles from './Footer.module.css';

const Footer = () => {
  return (
    <footer className={`py-5 ${styles.footer}`}>
      <div className="container-lg">
        <div className="d-flex flex-row flex-wrap align-items-stretch justify-content-between text-light">
          <div className={styles.footerCol}>
            <div className="h6 fw-bold text-uppercase">Project</div>
            <div className="d-flex flex-column text-start">
              <ExternalLink
                className="mb-1 opacity-75"
                href="https://github.com/cncf/clomonitor/tree/main/docs"
                label="Open documentation"
              >
                Documentation
              </ExternalLink>
            </div>
          </div>

          <div className={styles.footerCol}>
            <div className="h6 fw-bold text-uppercase">Community</div>
            <div className="d-flex flex-column text-start">
              <ExternalLink className="mb-1 opacity-75" href="https://github.com/cncf/clomonitor" label="Open Github">
                <div className="d-flex align-items-center">
                  <FaGithub className="me-2" />
                  GitHub
                </div>
              </ExternalLink>
            </div>
          </div>

          <div className={styles.footerCol}>
            <div className="h6 fw-bold text-uppercase">About</div>
            <div className={`opacity-75 ${styles.license}`}>
              CLOMonitor is an <b className="d-inline-block">Open Source</b> project licensed under the{' '}
              <ExternalLink
                className="d-inline-block mb-1"
                href="https://www.apache.org/licenses/LICENSE-2.0"
                label="Open Apache License 2.0 documentation"
              >
                <div className="d-flex align-items-center">
                  Apache License 2.0
                  <FiExternalLink className={`ms-1 ${styles.miniIcon}`} />
                </div>
              </ExternalLink>
            </div>
          </div>

          <div className={`ms-0 ms-lg-auto ${styles.fullMobileSection}`}>
            <img className={styles.logo} alt="Logo" src={logo} />
          </div>
        </div>
      </div>
    </footer>
  );
};

export default Footer;
