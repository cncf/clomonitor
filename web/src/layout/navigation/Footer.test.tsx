import { render, screen } from '@testing-library/react';

import Footer from './Footer';

describe('Footer', () => {
  afterEach(() => {
    jest.resetAllMocks();
  });

  it('creates snapshot', () => {
    const { asFragment } = render(<Footer />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('renders proper content', () => {
    render(<Footer />);

    expect(screen.getByText('Project')).toBeInTheDocument();
    expect(screen.getByText('Community')).toBeInTheDocument();
    expect(screen.getByText('About')).toBeInTheDocument();
    expect(screen.getByText(/CLOMonitor is an/)).toBeInTheDocument();
    expect(screen.getByText('Open Source')).toBeInTheDocument();

    const links = screen.getAllByRole('link');
    expect(links).toHaveLength(3);
    expect(links[0]).toHaveTextContent('Documentation');
    expect(links[0]).toHaveAttribute('href', 'https://github.com/cncf/clomonitor/tree/main/docs');
    expect(links[1]).toHaveTextContent('GitHub');
    expect(links[1]).toHaveAttribute('href', 'https://github.com/cncf/clomonitor');
    expect(links[2]).toHaveTextContent('Apache License 2.0');
    expect(links[2]).toHaveAttribute('href', 'https://www.apache.org/licenses/LICENSE-2.0');
  });
});
