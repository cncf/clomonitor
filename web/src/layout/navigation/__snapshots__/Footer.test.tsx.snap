// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer creates snapshot 1`] = `
<DocumentFragment>
  <footer
    class="py-5 footer"
  >
    <div
      class="container-lg"
    >
      <div
        class="d-flex flex-row flex-wrap align-items-stretch justify-content-between text-light"
      >
        <div
          class="footerCol"
        >
          <div
            class="h6 fw-bold text-uppercase"
          >
            Project
          </div>
          <div
            class="d-flex flex-column text-start"
          >
            <a
              aria-label="Open documentation"
              class="link mb-1 opacity-75"
              href="https://github.com/cncf/clomonitor/tree/main/docs"
              rel="noopener noreferrer"
              tabindex="-1"
              target="_blank"
            >
              <div
                class="d-flex flex-row align-items-baseline"
              >
                Documentation
              </div>
            </a>
          </div>
        </div>
        <div
          class="footerCol"
        >
          <div
            class="h6 fw-bold text-uppercase"
          >
            Community
          </div>
          <div
            class="d-flex flex-column text-start"
          >
            <a
              aria-label="Open Github"
              class="link mb-1 opacity-75"
              href="https://github.com/cncf/clomonitor"
              rel="noopener noreferrer"
              tabindex="-1"
              target="_blank"
            >
              <div
                class="d-flex flex-row align-items-baseline"
              >
                <div
                  class="d-flex align-items-center"
                >
                  <svg
                    class="me-2"
                    fill="currentColor"
                    height="1em"
                    stroke="currentColor"
                    stroke-width="0"
                    viewBox="0 0 496 512"
                    width="1em"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M165.9 397.4c0 2-2.3 3.6-5.2 3.6-3.3.3-5.6-1.3-5.6-3.6 0-2 2.3-3.6 5.2-3.6 3-.3 5.6 1.3 5.6 3.6zm-31.1-4.5c-.7 2 1.3 4.3 4.3 4.9 2.6 1 5.6 0 6.2-2s-1.3-4.3-4.3-5.2c-2.6-.7-5.5.3-6.2 2.3zm44.2-1.7c-2.9.7-4.9 2.6-4.6 4.9.3 2 2.9 3.3 5.9 2.6 2.9-.7 4.9-2.6 4.6-4.6-.3-1.9-3-3.2-5.9-2.9zM244.8 8C106.1 8 0 113.3 0 252c0 110.9 69.8 205.8 169.5 239.2 12.8 2.3 17.3-5.6 17.3-12.1 0-6.2-.3-40.4-.3-61.4 0 0-70 15-84.7-29.8 0 0-11.4-29.1-27.8-36.6 0 0-22.9-15.7 1.6-15.4 0 0 24.9 2 38.6 25.8 21.9 38.6 58.6 27.5 72.9 20.9 2.3-16 8.8-27.1 16-33.7-55.9-6.2-112.3-14.3-112.3-110.5 0-27.5 7.6-41.3 23.6-58.9-2.6-6.5-11.1-33.3 2.6-67.9 20.9-6.5 69 27 69 27 20-5.6 41.5-8.5 62.8-8.5s42.8 2.9 62.8 8.5c0 0 48.1-33.6 69-27 13.7 34.7 5.2 61.4 2.6 67.9 16 17.7 25.8 31.5 25.8 58.9 0 96.5-58.9 104.2-114.8 110.5 9.2 7.9 17 22.9 17 46.4 0 33.7-.3 75.4-.3 83.6 0 6.5 4.6 14.4 17.3 12.1C428.2 457.8 496 362.9 496 252 496 113.3 383.5 8 244.8 8zM97.2 352.9c-1.3 1-1 3.3.7 5.2 1.6 1.6 3.9 2.3 5.2 1 1.3-1 1-3.3-.7-5.2-1.6-1.6-3.9-2.3-5.2-1zm-10.8-8.1c-.7 1.3.3 2.9 2.3 3.9 1.6 1 3.6.7 4.3-.7.7-1.3-.3-2.9-2.3-3.9-2-.6-3.6-.3-4.3.7zm32.4 35.6c-1.6 1.3-1 4.3 1.3 6.2 2.3 2.3 5.2 2.6 6.5 1 1.3-1.3.7-4.3-1.3-6.2-2.2-2.3-5.2-2.6-6.5-1zm-11.4-14.7c-1.6 1-1.6 3.6 0 5.9 1.6 2.3 4.3 3.3 5.6 2.3 1.6-1.3 1.6-3.9 0-6.2-1.4-2.3-4-3.3-5.6-2z"
                    />
                  </svg>
                  GitHub
                </div>
              </div>
            </a>
          </div>
        </div>
        <div
          class="footerCol"
        >
          <div
            class="h6 fw-bold text-uppercase"
          >
            About
          </div>
          <div
            class="opacity-75 license"
          >
            CloMonitor is an 
            <b
              class="d-inline-block"
            >
              Open Source
            </b>
             project licensed under the 
            <a
              aria-label="Open Apache License 2.0 documentation"
              class="link d-inline-block mb-1"
              href="https://www.apache.org/licenses/LICENSE-2.0"
              rel="noopener noreferrer"
              tabindex="-1"
              target="_blank"
            >
              <div
                class="d-flex flex-row align-items-baseline"
              >
                <div
                  class="d-flex align-items-center"
                >
                  Apache License 2.0
                  <svg
                    class="ms-1 miniIcon"
                    fill="none"
                    height="1em"
                    stroke="currentColor"
                    stroke-linecap="round"
                    stroke-linejoin="round"
                    stroke-width="2"
                    viewBox="0 0 24 24"
                    width="1em"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M18 13v6a2 2 0 0 1-2 2H5a2 2 0 0 1-2-2V8a2 2 0 0 1 2-2h6"
                    />
                    <polyline
                      points="15 3 21 3 21 9"
                    />
                    <line
                      x1="10"
                      x2="21"
                      y1="14"
                      y2="3"
                    />
                  </svg>
                </div>
              </div>
            </a>
          </div>
        </div>
        <div
          class="ms-0 ms-lg-auto fullMobileSection"
        >
          <img
            alt="Logo"
            class="logo"
            src=""
          />
        </div>
      </div>
    </div>
  </footer>
</DocumentFragment>
`;
